<?xml version="1.0"?>
<robot name="my_robot" xmlns:xacro="http://www.ros.org/wiki/xacro" >

<xacro:property name="platform height" value="0.2" />
<xacro:property name="body" value="0.6" />


    <material name="grey">
        <color rgba="0.7 0.7 0.7 1.0"/>
    </material>

    <material name="green">
        <color rgba="0.0 0.6 0.0 1"/>
    </material>

    <material name="white">
        <color rgba="1.0 1.0 1.0 1.0"/>
    </material>

    <material name="blue">
        <color rgba="0.15 0.2 0.4 1.0"/>
    </material>

    <!-- Base of the robot -->
    <link name="base_link">
        <visual>
            <geometry>
                <box size="0.2 0.2 0.2"/>
            </geometry>
            <origin xyz="0.0 0.0 0.20" rpy="0.0 0.0 0.0"/>
            <material name="green"/>
        </visual>
    </link>

    <link name="pole_link">
        <visual>
            <geometry>
                    <cylinder radius="0.025" length="0.4"/>
            </geometry>
            <origin xyz="0.0 0.0 0.4" rpy="0.0 0.0 0.0"/>
            <material name="grey"/>
        </visual>
    </link>

    <joint name="pole_to_base_joint" type="fixed">
        <parent link="base_link"/>
        <child link="pole_link"/>
        <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </joint>

    
    <link name="platform">
        <visual>
            <geometry>
                 <cylinder radius="0.3" length="0.05"/>
            </geometry>
             <origin xyz="0.0 0.0 0.6" rpy="0.0 0.0 0.0"/>
             <material name="blue"/> 
        </visual>
    </link>

    <joint name="top_wheel_joint" type="fixed">
        <parent link="pole_link"/>
        <child link="platform"/>
        <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </joint>


    <!-- Base footprint link -->
    <link name="base_footprint"/>
    
    <!-- Base footprint joint -->
    <joint name="base_joint" type="fixed">
        <parent link="base_footprint"/>
        <child link="base_link"/>
        <origin xyz="0.0 0.0 0.1" rpy="0.0 0.0 0.0"/>
    </joint>



    <!-- Caster wheel link -->
    <link name="wheel">
        <visual>
            <geometry>
                 <cylinder radius="0.1" length="0.05"/>
            </geometry>
             <origin xyz="0.0 0.0 0.0" rpy="1.57 0.0 0.0"/>       
        </visual>
    </link>

    <!-- Caster wheel joint -->
    <joint name="wheel_joint" type="continuous">
        <parent link="base_link"/>
        <child link="wheel"/>
        <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
        <axis xyz="0.0 1.0 0.0"/>
    </joint>
</robot>
